contrib/log_correlation/stackdriver/src/test/java/io/opencensus/contrib/logcorrelation/stackdriver/OpenCensusTraceLoggingEnhancerTest.java

```diff
@@ -49,53 +55,119 @@
   private static final Tracer tracer = Tracing.getTracer();
 
   @Test
-  public void enhanceLogEntry_Sampled() {
+  public void enhanceLogEntry_DoNotAddSampledSpanToLogEntryWithNoSpans() {
     LogEntry logEntry =
         getEnhancedLogEntry(
             "my-test-project-1",
+            new OpenCensusTraceLoggingEnhancer(SpanSelection.NO_SPANS),
+            new TestSpan(
+                SpanContext.create(
+                    TraceId.fromLowerBase16("3da31be987098abb08c71c7700d2680e"),
+                    SpanId.fromLowerBase16("51b109f15e0d3881"),
+                    TraceOptions.builder().setIsSampled(true).build())));
+    assertContainsNoTracingData(logEntry);
+  }
+
+  @Test
+  public void enhanceLogEntry_AddSampledSpanToLogEntryWithSampledSpans() {
+    LogEntry logEntry =
+        getEnhancedLogEntry(
+            "my-test-project-2",
+            new OpenCensusTraceLoggingEnhancer(SpanSelection.SAMPLED_SPANS),
             new TestSpan(
                 SpanContext.create(
                     TraceId.fromLowerBase16("4c9874d0b41224cce77ff74ee10f5ee6"),
                     SpanId.fromLowerBase16("592ae363e92cb3dd"),
                     TraceOptions.builder().setIsSampled(true).build())));
-    assertThat(logEntry.getLabels().get("sampled")).isEqualTo("true");
+    assertThat(logEntry.getLabels()).containsEntry("sampled", "true");
     assertThat(logEntry.getTrace())
-        .isEqualTo("projects/my-test-project-1/traces/4c9874d0b41224cce77ff74ee10f5ee6");
+        .isEqualTo("projects/my-test-project-2/traces/4c9874d0b41224cce77ff74ee10f5ee6");
     assertThat(logEntry.getSpanId()).isEqualTo("592ae363e92cb3dd");
   }
 
   @Test
-  public void enhanceLogEntry_NotSampled() {
+  public void enhanceLogEntry_AddSampledSpanToLogEntryWithAllSpans() {
     LogEntry logEntry =
         getEnhancedLogEntry(
-            "my-test-project-2",
+            "my-test-project-3",
+            new OpenCensusTraceLoggingEnhancer(SpanSelection.ALL_SPANS),
+            new TestSpan(
+                SpanContext.create(
+                    TraceId.fromLowerBase16("4c6af40c499951eb7de2777ba1e4fefa"),
+                    SpanId.fromLowerBase16("de52e84d13dd232d"),
+                    TraceOptions.builder().setIsSampled(true).build())));
```

Comment:
Shouldn't you set this to false in order to check that it's collected regardless?

Reply:
Misread the name - ignore.